src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -1222,486 +1631,614 @@ s32 func_80A13B1C(EnGo* this, PlayState* play) {
+    } else if (this->actionFlags & ENGO_FLAG_UNROLLING) {
         if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
-            this->unk_390 |= 0x80;
-            this->unk_390 &= ~0x8000;
+            this->actionFlags |= ENGO_FLAG_STANDING;
+            this->actionFlags &= ~ENGO_FLAG_UNROLLING;
         }
-    } else if (this->unk_390 & 0x200) {
-        if ((this->actor.xzDistToPlayer < 160.0f) && (this->actor.playerHeightRel < 20.0f) && (this->unk_3EC == 0)) {
-            func_80A12C48(this, play, 3);
-            this->unk_390 &= ~0x80;
-            this->unk_390 &= ~0x200;
-            this->unk_390 |= 0x8000;
-            this->actor.shape.yOffset = 0.0f;
-        } else if ((this->unk_3EC != 0) && (gSaveContext.save.weekEventReg[22] & 4)) {
-            this->actor.scale.x = this->unk_3A4 - (Math_SinS(this->unk_3AE) * 0.001f);
-            this->actor.scale.y = (Math_SinS(this->unk_3AE) * 0.001f) + this->unk_3A4;
-            this->actor.scale.z = (Math_SinS(this->unk_3AE) * 0.001f) + this->unk_3A4;
-            if (this->unk_3AE == 0) {
-                this->unk_3EC = -this->unk_3EC;
-                if (this->unk_3EC > 0) {
-                    sfxId = NA_SE_EN_GOLON_SNORE1;
-                } else {
-                    sfxId = NA_SE_EN_GOLON_SNORE2;
-                }
+    } else if (this->actionFlags & ENGO_FLAG_ROLLEDUP) {
+        if ((this->actor.xzDistToPlayer < 160.0f) && (this->actor.playerHeightRel < 20.0f) &&
+            (this->sleepState == ENGO_AWAKE)) {
+            EnGo_ChangeAnim(this, play, ENGO_ANIM_UNROLL_IMM);
+            this->actionFlags &= ~ENGO_FLAG_STANDING;
+            this->actionFlags &= ~ENGO_FLAG_ROLLEDUP;
+            this->actionFlags |= ENGO_FLAG_UNROLLING;
+            this->actor.shape.yOffset = ENGO_STANDING_Y_OFFSET;
+        } else if ((this->sleepState != ENGO_AWAKE) &&
+                   (gSaveContext.save.weekEventReg[22] & 0x4 /* Elder's son has been calmed */)) {
+            // While asleep, rhythmicallly snore and distort shape to show breathing.
+            this->actor.scale.x = this->scaleFactor - (Math_SinS(this->snorePhase) * (0.1f * ENGO_NORMAL_SCALE));
+            this->actor.scale.y = this->scaleFactor + (Math_SinS(this->snorePhase) * (0.1f * ENGO_NORMAL_SCALE));
+            this->actor.scale.z = this->scaleFactor + (Math_SinS(this->snorePhase) * (0.1f * ENGO_NORMAL_SCALE));
+
+            if (this->snorePhase == 0) {
+                this->sleepState = -this->sleepState;
+                sfxId = (this->sleepState > ENGO_AWAKE) ? NA_SE_EN_GOLON_SNORE1 : NA_SE_EN_GOLON_SNORE2;
                 Actor_PlaySfxAtPos(&this->actor, sfxId);
             }
-            this->unk_3AE += 0x400;
-            this->actor.shape.yOffset = (this->actor.scale.y / this->unk_3A4) * 14.0f;
-            SubS_UpdateFlags(&this->unk_390, 3, 7);
+            this->snorePhase += 0x400; // ~5.7 degrees per frame or ~1/3 rotation per second.
```

Comment:
I don't think talking about degrees in what is essentially a phase is useful. It's also not really a rotation.